### ISSUE TITLE
Compile Stats Learning Outcomes

### ISSUE THREAD

**Issue by firasm** [open-resources/learning_outcomes#37]
The textbook we'll be using for the Stats Problem bank is : https://github.com/OpenIntroStat

The Learning Outcomes are available as .tex and .pdf files, [for example this one](https://github.com/OpenIntroStat/openintro-statistics-learn-obj/blob/master/01/01learn_obj.tex)

![Screenshot 2023-06-07 at 11 02 42 AM](https://github.com/open-resources/learning_outcomes/assets/2507459/bcdf3090-ce2d-43d3-bc0b-aca751f21d11)

![Screenshot 2023-06-07 at 11 02 31 AM](https://github.com/open-resources/learning_outcomes/assets/2507459/9ae171f9-2fb6-4587-92f5-cb4c9d48bcfe)

The task is to write a script that pulls everything with `\item` from that git repo with file names `0Xlearn_obj.tex` and try to generate a list of learning outcomes by Chapter and Topic.

**Comment by firasm** [comment]
We want to generate a `yml` file similar to this:

https://github.com/open-resources/learning_outcomes/blob/main/Masterlist_Topics_Outcomes.yml

I suggest focusing on the extraction part, I can help you with the yml export

**Comment by alebuiles** [comment]
@firasm, I have just created a branch and uploaded a Python script to address the compilation of learning outcomes. I wanted to confirm how I should name the file and whether the current solution aligns with our requirements (it is the first draft). By now, the script is printing a list of all the possible \item from the .tex files. However, it does not organize the outcomes by topic, chapter, or subtopic. Should I modify the script to include this hierarchical structure, or is it intended for the `.yml` file only?

I appreciate your guidance and feedback on these questions.

**Comment by firasm** [comment]
Hi @alebuiles I haven't had a chance to look at the script yet, but if you can easily organize it so that it will print by topic (chapter) at least, it will make the manual work of sorting it later much faster (to produce the .yml file). 

Remember the goal is to get it into the form of the Physics Learning Outcomes.

**Comment by alebuiles** [comment]
Hello @firasm! 

I wanted to inform you that I have made some updates to the script. It is now organized by chapter, making it easier to produce the YAML file. Additionally, I have created the YAML file to be similar to the physics learning outcomes, and I have copied the output directly from the script.

Let me know if this meets your expectations or what changes should I do either to the script or the .yml file. Thank you!
